sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1563,6 +1564,17 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val DEFAULT_CACHE_STORAGE_LEVEL = buildConf("spark.sql.defaultCacheStorageLevel")
+    .doc("The default storage level of `dataset.cache()`, `catalog.cacheTable()` and " +
+      "sql query `CACHE TABLE t`.")
+    .version("4.0.0")
+    .stringConf
+    .transform(_.toUpperCase(Locale.ROOT))
+    .checkValues(Set("NONE", "DISK_ONLY", "DISK_ONLY_2", "DISK_ONLY_3", "MEMORY_ONLY",
+      "MEMORY_ONLY_2", "MEMORY_ONLY_SER", "MEMORY_ONLY_SER_2", "MEMORY_AND_DISK",
+      "MEMORY_AND_DISK_2", "MEMORY_AND_DISK_SER", "MEMORY_AND_DISK_SER_2", "OFF_HEAP"))
```

Comment:
Because `StorageLevel.fromString` matches the string value, it seems we can avoid copy these string literal and call `checkValues` here.

cc @cloud-fan

Reply:
this is for better error msg if user set a wrong storage level

Reply:
Hears good to me!

Reply:
https://github.com/apache/spark/pull/43278 merged. You can use `StorageLevelMapper.values().map(_.name()).mkString("\"", ", ", "\"")` now.